inc/saiswitch.h

```diff
@@ -1113,98 +1236,73 @@ typedef enum _sai_switch_attr_t
  *
  * Adapter DLL may request a shutdown due to an unrecoverable failure
  * or a maintenance operation
+ *
+ * @param[in] switch_id Switch Id
  */
-typedef void (*sai_switch_shutdown_request_fn)(void);
+typedef void (*sai_switch_shutdown_request_fn)(
+              _In_ sai_object_id_t switch_id);
 
 /**
  * @brief Switch oper state change notification
  *
+ * @param[in] switch_id Switch Id
  * @param[in] switch_oper_status New switch oper state
  */
 typedef void (*sai_switch_state_change_notification_fn)(
+        _In_ sai_object_id_t switch_id,
         _In_ sai_switch_oper_status_t switch_oper_status);
 
 /**
- * @brief Switch notification table passed to the adapter via sai_initialize_switch()
- */
-typedef struct _sai_switch_notification_t
-{
-    sai_switch_state_change_notification_fn on_switch_state_change;
-    sai_fdb_event_notification_fn           on_fdb_event;
-    sai_port_state_change_notification_fn   on_port_state_change;
-    sai_port_event_notification_fn          on_port_event;
-    sai_switch_shutdown_request_fn          on_switch_shutdown_request;
-    sai_packet_event_notification_fn        on_packet_event;
-
-} sai_switch_notification_t;
-
-/**
- * @brief SDK initialization. After the call the capability attributes should
- * be ready for retrieval via sai_get_switch_attribute().
+ * @brief Create switch
  *
- * @param[in] profile_id Handle for the switch profile.
- * @param[in] switch_hardware_id Switch hardware ID to open
- * @param[in] firmware_path_name Vendor specific path name of the firmware to load
- * @param[in] switch_notifications Switch notification table
+ *   SDK initialization/connect to SDK. After the call the capability attributes should be
+ *   ready for retrieval via sai_get_switch_attribute().
  *
- * @return #SAI_STATUS_SUCCESS on success Failure status code on error
- */
-typedef sai_status_t (*sai_initialize_switch_fn)(
-        _In_ sai_switch_profile_id_t profile_id,
-        _In_reads_z_(SAI_MAX_HARDWARE_ID_LEN) char *switch_hardware_id,
-        _In_reads_opt_z_(SAI_MAX_FIRMWARE_PATH_NAME_LEN) char *firmware_path_name,
-        _In_ sai_switch_notification_t *switch_notifications);
-
-/**
- * @brief Release all resources associated with currently opened switch
+ * @param[out] switch_id The Switch Object ID
+ * @param[in] attr_count number of attributes
+ * @param[in] attr_list Array of attributes
  *
- * @param[in] warm_restart_hint Hint that indicates controlled warm restart.
- * Since warm restart can be caused by crash (therefore there are no guarantees
- * for this call), this hint is really a performance optimization.
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
  */
-typedef void (*sai_shutdown_switch_fn)(
-        _In_ bool warm_restart_hint);
+typedef sai_status_t(*sai_create_switch_fn)(
```

Comment:
Before creating new switch instance using sai_create_switch_fn(), how application should get available NPU numbers for particular hardware?
For example, application wants to initialize two NPU's, but particular platform is single NPU? Can application retrieve NPU number prior to create new switch instances?